ydb/core/tx/schemeshard/schemeshard_audit_log.cpp

```diff
@@ -167,6 +178,209 @@ void AuditLogModifySchemeTransactionDeprecated(const NKikimrScheme::TEvModifySch
     }
 }
 
+namespace {
+
+struct TXxportRecord {
+    TString OperationName;
+    ui64 TxId;
+    TString Uid;
+    TString RemoteAddress;
+    TString UserSID;
+    TString DatabasePath;
+    TString Status;
+    Ydb::StatusIds::StatusCode DetailedStatus;
+    TString Reason;
+    TVector<TString> Paths;
+    TVector<std::pair<TString, TString>> AdditionalParts;
+    TString StartTime;
+    TString EndTime;
+    TString CloudId;
+    TString FolderId;
+    TString ResourceId;
+};
+
+void AuditLogXxport(TXxportRecord&& record) {
+    AUDIT_LOG(
+        AUDIT_PART("component", SchemeshardComponentName)
+
+        AUDIT_PART("tx_id", std::to_string(record.TxId))
+        AUDIT_PART("remote_address", (!record.RemoteAddress.empty() ? record.RemoteAddress : EmptyValue))
+        AUDIT_PART("subject", (!record.UserSID.empty() ? record.UserSID : EmptyValue))
+        AUDIT_PART("database", (!record.DatabasePath.empty() ? record.DatabasePath : EmptyValue))
+        AUDIT_PART("operation", record.OperationName)
+        AUDIT_PART("status", record.Status)
+        AUDIT_PART("detailed_status", Ydb::StatusIds::StatusCode_Name(record.DetailedStatus))
+        AUDIT_PART("reason", record.Reason)
+
+        AUDIT_PART("paths", RenderList(record.Paths), !record.Paths.empty())
+
+        AUDIT_PART("export_uid", record.Uid);
+
+        // all parts are considered required, so all empty values are replaced with a special stub
+        for (const auto& [name, value] : record.AdditionalParts) {
+            AUDIT_PART(name, (!value.empty() ? value : EmptyValue))
+        }
+
+        // AUDIT_PART("start_time", exportInfo.StartTime.ToString(), exportInfo.StartTime != TInstant::Zero())
+        // AUDIT_PART("end_time", exportInfo.EndTime.ToString(), exportInfo.EndTime != TInstant::Zero())
+        AUDIT_PART("start_time", record.StartTime)
+        AUDIT_PART("end_time", record.EndTime)
```

Comment:
Do we need to check `record.StartTime` and `record.EndTime` for emptiness and set default stub? These values don`t set in `*xportStart` function

Reply:
It is intentional. Start record shouldn't have times at all (it can have `start_time`, but why, and cannot have `end_time`). And end record must have both times.
And that is responsibility of `_AuditLogXxport{Start,End}()`